### ISSUE TITLE
--with-lustre fails   ERR("ior was not compiled with Lustre support")   yet linux/lustre/lustre_user.h is present

### ISSUE THREAD

**Issue by AeonJJohnson** [hpc/ior#189]
Greetings,

First, I'm willing to accept the answer that the problem here exists between the keyboard and chair.

ior (branches tried): master, 3.2

1. Edited configure to only look for linux/lustre/lustre_user.h (avoiding the deprecated file in lustre/lustre_user.h
2. ./bootstrap.sh && ./configure --prefix=/home/aeon/openmpi --with-lustre 

config.log shows lustre_user.h
```
configure:5462: checking linux/lustre/lustre_user.h usability
configure:5462: mpicc -c -g -O2  conftest.c >&5
configure:5462: $? = 0
configure:5462: result: yes
configure:5462: checking linux/lustre/lustre_user.h presence
configure:5462: mpicc -E  conftest.c
configure:5462: $? = 0
configure:5462: result: yes
configure:5462: checking for linux/lustre/lustre_user.h
configure:5462: result: yes
...
ac_cv_header_linux_lustre_lustre_user_h=yes
```
3. make
4. mpirun 
```
mpirun -np $NP -machinefile /home/aeon/machines /home/aeon/jeff/ior/src/ior -v -a POSIX -g -e -w -D 300 -b 16g -t 4m -o /testfs/pools/testfile -C -O lustreStripeCount=4
...
<<insert trainwreck here>>
ior ERROR: ior was not compiled with Lustre support, errno 2, No such file or directory (parse_options.c:259)
```
Same result with master and 3.2 branches. Same result with any or all Lustre specific options set. 

ior functions normally without "-O SomeLustreOption" set.

**Comment by glennklockwood** [comment]
Can confirm.  This is because we check for lustre_user.h in two places now, resulting in both

    /* Define to 1 if you have the <linux/lustre/lustre_user.h> header file. */
    #define HAVE_LINUX_LUSTRE_LUSTRE_USER_H 1
    
    /* Define to 1 if you have the <lustre/lustre_user.h> header file. */
    /* #undef HAVE_LUSTRE_LUSTRE_USER_H */

which is problematic when `parse_options.c` only checks for `#ifndef HAVE_LUSTRE_LUSTRE_USER_H`.  Will push a fix in a few minutes.

**Comment by glennklockwood** [comment]
https://github.com/hpc/ior/pull/new/fix-189 (based on master) should re-enable the Lustre stuff.  What an embarrassing bug.

Once you confirm that everything is working with this fix branch, I will backport it to rc and 3.2.  Maybe it's time to cut a new micro release for IOR-3.2 after that.

**Comment by AeonJJohnson** [comment]
So far so good. Accepts and executes with Lustre options. Going through various stripe options to check function. I'll report back after a few different runs with varied options.

Thanks!
